packages/compass-data-modeling/src/store/export-diagram.ts

```diff
@@ -0,0 +1,181 @@
+import type { Reducer } from 'redux';
+import { isAction } from './util';
+import type { DataModelingThunkAction } from './reducer';
+import { exportToJson, exportToPng } from '../services/export-diagram';
+import { getCurrentDiagramFromState, selectCurrentModel } from './diagram';
+import { openToast } from '@mongodb-js/compass-components';
+import { isCancelError } from '@mongodb-js/compass-utils';
+import type { DiagramInstance } from '@mongodb-js/diagramming';
+
+export type ExportDiagramFormat = 'png' | 'json';
+
+export type ExportDiagramState = {
+  isModalOpen: boolean;
+  isExporting: boolean;
+  exportFormat?: ExportDiagramFormat;
+};
+
+export enum ExportDiagramActionTypes {
+  MODAL_OPENED = 'data-modeling/export-diagram/MODAL_OPENED',
+  MODAL_CLOSED = 'data-modeling/export-diagram/MODAL_CLOSED',
+  FORMAT_SELECTED = 'data-modeling/export-diagram/FORMAT_SELECTED',
+  EXPORT_STARTED = 'data-modeling/export-diagram/EXPORT_STARTED',
+  EXPORT_COMPLETED = 'data-modeling/export-diagram/EXPORT_COMPLETED',
+}
+
+type ModalOpenedAction = {
+  type: ExportDiagramActionTypes.MODAL_OPENED;
+};
+
+type ModalClosedAction = {
+  type: ExportDiagramActionTypes.MODAL_CLOSED;
+};
+
+type FormatSelectedAction = {
+  type: ExportDiagramActionTypes.FORMAT_SELECTED;
+  format: ExportDiagramFormat;
+};
+
+type ExportStartedAction = {
+  type: ExportDiagramActionTypes.EXPORT_STARTED;
+};
+
+type ExportCompletedAction = {
+  type: ExportDiagramActionTypes.EXPORT_COMPLETED;
+};
+
+export type ExportDiagramActions =
+  | ModalOpenedAction
+  | ModalClosedAction
+  | FormatSelectedAction
+  | ExportStartedAction
+  | ExportCompletedAction;
+
+const INITIAL_STATE = {
+  isModalOpen: false,
+  isExporting: false,
+};
+
+let cancelExportAbortController: AbortController | null = null;
```

Comment:
This should be scoped to the plugin store, not global, extra thunk arg is how we usually do this

Reply:
added in [ae0d002](https://github.com/mongodb-js/compass/pull/7092/commits/ae0d002807b39ae1d457791ad9612a4f798585c9)